### DISCUSSION TITLE
useFieldArray not saving data in the form

### DISCUSSION THREAD

**Discussion by EmanueleCiriachi** [react-hook-form/react-hook-form#5882] [Q&A] [answered]
I created the following CodeSandbox:
https://codesandbox.io/s/zealous-roentgen-8di5x?file=/src/App.tsx

Where I have a `useForm` containing an Array displayed with a dynamically extensible `useFieldArray`, that features an initial value.
I have two problems:

1) my initial value is not reflected in the form, and
2) If I add multiple entries to this array, those values are not saved (this can be seen via the `watch` directive in my CodeSandbox).

What am I doing wrong?

**Comment by bluebill1049** [comment, marked as answer]
> my initial value is not reflected in the form, and

you never did register any inputs

<img width="605" alt="Screen Shot 2021-07-09 at 11 53 52 am" src="https://user-images.githubusercontent.com/10513364/125011719-56c9e200-e0ac-11eb-9485-38b7ea39d033.png">

**Reply by EmanueleCiriachi** [reply]
I see, thank you. I think I figured it out now... sort of:

https://codesandbox.io/s/dawn-shadow-cqu5x

I solved point 2.
However the first record now is always the default one, is not overwritten by what the user adds.
If you add more than 1 entry, then 2nd, 3rd, etc. will display the values inserted by the user, but the 1st will always retain the defaultValues (that are not shown anyway when you load the form).
